src/Notifier.js

```diff
@@ -133,29 +153,35 @@ var Notifier = {
         }
 
         if(enable) {
-            if (!this.havePermission()) {
-                global.Notification.requestPermission(function() {
-                    if (callback) {
-                        callback();
+            // Case when we do not have the permission as 'granted'
+            if (this.isPermissionDefault()) {
```

Comment:
I don't understand your answer.

`setEnabled` is not just used from the toolbar; it is also used from the settings dialog.

But the point is, you do not need to test for `isPermissionDefault` here - just go straight into requestPermission, which will give the same result.